test/api/object_lifeline_test.go

```diff
@@ -101,7 +127,18 @@ func TestLifeline_amendRecords(t *testing.T) {
 	err := heavymock.ImportRecords(ts.ConMngr.ImporterClient, allRecords)
 	require.NoError(t, err)
 
-	ts.WaitRecordsCount(t, count, 1000)
+	ts.BE.PulseClient.NextFinalizedPulseFunc = func(ctx context.Context, in *exporter.GetNextFinalizedPulse, opts ...grpc.CallOption) (*exporter.FullPulse, error) {
+		p := uint32(ts.ConMngr.Importer.GetLowestUnsentPulse())
+		if p == 1<<32-1 {
+			return nil, errors.New("unready yet")
+		}
+		return clients.GetFullPulse(p), nil
```

Comment:
could you pls extract to the function and reuse in all places?

Reply:
as part of what ?

Reply:
agree, the func repeats all the time, should be reused

Reply:
ok, but we still do not know part of what it is.